### PR TITLE
feat(dc/gateway): add new resource support

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
Add new resource support.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. support virtual gateway resource.
2. support related document and acc test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/dc' TESTARGS='-run=TestAcVirtualGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dc -v -run=TestAcVirtualGateway_basic -timeout 360m -parallel 4
=== RUN   TestAcVirtualGateway_basic
=== PAUSE TestAcVirtualGateway_basic
=== CONT  TestAcVirtualGateway_basic
--- PASS: TestAcVirtualGateway_basic (40.85s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc        40.903s
```

### CODE REVIEW

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dc' TESTARGS='-run=TestAcVirtualGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dc -v -run=TestAcVirtualGateway_basic -timeout 360m -parallel 4
=== RUN   TestAcVirtualGateway_basic
=== PAUSE TestAcVirtualGateway_basic
=== CONT  TestAcVirtualGateway_basic
--- PASS: TestAcVirtualGateway_basic (40.76s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc        40.859s
```

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dc' TESTARGS='-run=TestAccVirtualGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dc -v -run=TestAccVirtualGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccVirtualGateway_basic
=== PAUSE TestAccVirtualGateway_basic
=== CONT  TestAccVirtualGateway_basic
--- PASS: TestAccVirtualGateway_basic (46.58s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc        46.673s
```

File: docs/resources/dc_virtual_gateway.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+* `region` - (Optional, String, ForceNew) Specifies the region where the virtual gateway is located.  
+  If omitted, the provider-level region will be used. Changing this will create a new resource.
+
+* `vpc_id` - (Required, String, ForceNew) Specifies the ID of the VPC connected to the virtual gateway.  
+  Changing this will create a new resource.
+
+* `local_ep_group` - (Required, List) Specifies the list of IPv6 subnets from the virtual gateway to access cloud
+  services, which is usually the CIDR block of the VPC.
+
+* `name` - (Optional, String) Specifies the name of the virtual gateway.
```
Comment (ShiChangkuo) [inline docs/resources/dc_virtual_gateway.md]: it's better to mark `name` be **required**.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/dc' TESTARGS='-run=TestAccVirtualGateway_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dc -v -run=TestAccVirtualGateway_basic -timeout 360m -parallel 4
=== RUN   TestAccVirtualGateway_basic
=== PAUSE TestAccVirtualGateway_basic
=== CONT  TestAccVirtualGateway_basic
--- PASS: TestAccVirtualGateway_basic (41.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dc        42.042s
```

**Pull request merged**